### PR TITLE
Patch: Better error reporting when a watched path does not exist

### LINKED ISSUE #312: "run_ghci: ghci event channel closed" and "run_watcher: No such file or directory (os error 2)"
### What happened?

I am trying to run ghciwatch inside a docker container, but I get the error message:

```
$ ~/.local/bin/ghciwatch --command="cabal repl my-project"
Error:   × Tasks failed:
  │ • run_ghci: ghci event channel closed
  │ • run_watcher: No such file or directory (os error 2)
  ```

### What did you expect to happen?

I expect ghciwatch to work

### Steps to reproduce the issue

Unfortunately this is on a proprietary codebase. There does not seem to be a verbose option that I can enable to get more logs.

### The version of ghciwatch with the bug

ghciwatch 1.0.1

### PR DESCRIPTION
**Pull request by Xophmeister**
Provide a more helpful error message when a watched path does not exist. For example,

```console
$ ghciwatch --watch does-not-exist
```

**Before**
```
Error:   × Tasks failed:
  │ • run_ghci: ghci event channel closed
  │ • run_watcher: No such file or directory (os error 2)
```

**After**
```
Error:   × Tasks failed:
  │ • run_ghci: ghci event channel closed
  │ • run_watcher: Cannot watch path that doesn't exist: "/home/chris/Projects/tweag/ghciwatch/does-not-exist"
```

Partially resolves #312

- [x] Labeled the PR with `patch`, `minor`, or `major` to request a version bump when it's merged.
- [ ] Updated the user manual in `docs/`.
- [ ] Added integration / regression tests in `tests/`.

### CODE REVIEW

**Comment by smelc** [comment]
@9999years> any feedback you would have on this PR? (so that we know if there is a chance it gets merged, and if so we'll work on reaching a final state (add a test?)).

File: src/watcher.rs [hunk 1]
```diff
@@ -96,7 +96,18 @@ async fn run_debouncer<T: notify::Watcher>(
...
-                .into_diagnostic()?;
+                .map_err(|err| match err {
+                    notify::Error {
+                        kind: notify::ErrorKind::Io(e),
+                        ..
+                    } if e.kind() == std::io::ErrorKind::NotFound => {
+                        miette::MietteDiagnostic::new(format!(
+                            "Cannot watch path that doesn't exist: {:?}",
+                            path.absolute()
+                        ))
```
Comment (9999years) [inline src/watcher.rs:107]: (Needs `use miette::miette;` at the top of the file.)

```suggestion
                        miette!(
                            "Cannot watch path that doesn't exist: {:?}",
                            path.absolute()
                        )
```

**Comment by 9999years** [comment]
@Xophmeister @smelc This is awesome, sorry I didn't see it earlier! I've been heads-down working on upgrading our monolith to GHC so my notifications have been stalled. Looks good! Left a couple comments but otherwise this should be a simple fix.

You might be able to write a test for it but the testing framework is a little clunky. Not required.

File: src/watcher.rs [hunk 2]
```diff
@@ -96,7 +96,18 @@ async fn run_debouncer<T: notify::Watcher>(
...
+                    }
+                    e => miette::MietteDiagnostic::new(e.to_string()),
```
[the remark this answers is not in the document]
Comment (Xophmeister) [reply src/watcher.rs]: Done in 7bc0e60

File: src/watcher.rs (hunk 1 above)
Comment (Xophmeister) [reply src/watcher.rs]: Done in 7bc0e60

**Comment by Xophmeister** [comment]
Thanks, @9999years :pray:

The original issue (#312) mentions two actions items:

> Action items here:
> 
> * [x]  Better error message when the `--watch` path isn't found.
> 
> * [ ]  `--verbose` and/or `--debug` flags as shortcuts for `--log-filter ghciwatch=debug,info` or similar.

Would your preference be for this PR to close 312 (thus needing the second item), or is this enough as is?

**Comment by 9999years** [comment]
This will close #312, I've filed #331 for the `--verbose`/`--debug` flags as those are tangential.

**Comment by 9999years** [comment]
Aside, this part of the message is still really clunky and unhelpful. Don't fix it in this PR but I just want to flag this as something that still needs work / isn't user-friendly enough:

```
Error:   × Tasks failed:
  │ • run_ghci: ghci event channel closed
```

**Comment by 9999years** [comment]
Anything left before merge? I see it's still marked as a draft.

**Comment by smelc** [comment]
@Xophmeister> WDYT? Ready to merge in your opinion?

**Comment by smelc** [comment]
Squashed to have a nicer history.

**Comment by 9999years** [comment]
Released this in 1.0.2: https://github.com/MercuryTechnologies/ghciwatch/releases/tag/v1.0.2

Thanks again for the contribution!

**Pull request merged**